### ISSUE TITLE
[Help] hackEarlySend causes players to be in offline mode

### ISSUE THREAD

**Issue by bluefireoly** [OKTW-Network/FabricProxy-Lite#5]
Enabling `hackEarlySend` - because I am using LuckPerms - causes all players to connect as "offline" players.

(When `hackEarlySend` is set to false (without LuckPerms) everyone is in normal online mode as usual, this is why I think that enabling `hackEarlySend` is the cause.)

Am I overseeing something here, or is this a bug?

**Comment by bluefireoly** [comment]
Update: This mod seems to be incompatible with Cardboard.

**Comment by bluefireoly** [comment]
For a fix, you might have a look at these Mixins from Cardboard:

`ServerLoginNetworkHandler` (the class where the mixin for `hackEarlySend` is applied):
https://github.com/CardboardPowered/cardboard/blob/master/src/main/java/org/cardboardpowered/mixin/network/MixinServerLoginNetworkHandler.java

and possibly any of these other network mixins is related aswell:
https://github.com/CardboardPowered/cardboard/tree/master/src/main/java/org/cardboardpowered/mixin/network

**Comment by bluefireoly** [comment]
Even more specific: I think [this mixin](https://github.com/CardboardPowered/cardboard/blob/1227016a8eee1f36062d746078e5a62af7f4b081/src/main/java/org/cardboardpowered/mixin/network/MixinServerLoginNetworkHandler.java#L255) causes the issue.

Is it possible to fix this incompatibility?

**Comment by james58899** [comment]
> 
> 
> Even more specific: I think [this mixin](https://github.com/CardboardPowered/cardboard/blob/1227016a8eee1f36062d746078e5a62af7f4b081/src/main/java/org/cardboardpowered/mixin/network/MixinServerLoginNetworkHandler.java#L255) causes the issue.
> 
> Is it possible to fix this incompatibility? (Or is this maybe an issue which has to be fixed by Cardboard?)

Yes it need fixed by Cardboard, it skip FabricProxy auth progress.
I think this mixin is not necessary, maybe Cardboard can add config to allow disable it.

**Comment by bluefireoly** [comment]
cardboard issue:
https://github.com/CardboardPowered/cardboard/issues/209

**Comment by bluefireoly** [comment]
Mh just a guess: Couldn't you remove [this cancel call](https://github.com/OKTW-Network/FabricProxy-Lite/blob/396ec6d46a317863b63d9dd8187ec10284de4a48/src/main/java/one/oktw/mixin/hack/ServerLoginNetworkHandler_EarlySendPacket.java#L28), and then I will create a PR to Cardboard to execute cardboards logic AFTER the logic of FabricProxy-Lite (only possible if FabricProxy-Lite does not cancel immediately)?
